plugins/base/src/main/kotlin/translators/descriptors/DescriptorAccessorConventionUtil.kt

```diff
@@ -10,29 +10,93 @@ import org.jetbrains.kotlin.load.java.propertyNamesBySetMethodName
 
 internal data class DescriptorFunctionsHolder(
     val regularFunctions: List<FunctionDescriptor>,
-    val accessors: Map<PropertyDescriptor, List<FunctionDescriptor>>
+    val accessors: Map<PropertyDescriptor, DescriptorAccessorHolder>
 )
 
-internal fun splitFunctionsAndAccessors(
+internal data class DescriptorAccessorHolder(
+    val getter: FunctionDescriptor? = null,
+    val setter: FunctionDescriptor? = null
+)
+
+/**
+ * Separate regular Kotlin/Java functions and inherited Java accessors
+ * to properly display properties inherited from Java.
+ *
+ * Take this example:
+ * ```
+ * // java
+ * public class JavaClass {
+ *     private int a = 1;
+ *     public int getA() { return a; }
+ *     public void setA(int a) { this.a = a; }
+ * }
+ *
+ * // kotlin
+ * class Bar : JavaClass() {
+ *     fun foo() {}
+ * }
+ * ```
+ *
+ * It should result in:
+ * - 1 regular function `foo`
+ * - Map a=[`getA`, `setA`]
+ */
+internal fun splitFunctionsAndInheritedAccessors(
     properties: List<PropertyDescriptor>,
     functions: List<FunctionDescriptor>
 ): DescriptorFunctionsHolder {
+    val (javaMethods, kotlinFunctions) = functions.partition { it is JavaMethodDescriptor }
+    if (javaMethods.isEmpty()) {
+        return DescriptorFunctionsHolder(regularFunctions = kotlinFunctions, emptyMap())
+    }
+
     val propertiesByName = properties.associateBy { it.name.asString() }
-    val regularFunctions = mutableListOf<FunctionDescriptor>()
-    val accessors = mutableMapOf<PropertyDescriptor, MutableList<FunctionDescriptor>>()
-    functions.forEach { function ->
+    val regularFunctions = ArrayList<FunctionDescriptor>(kotlinFunctions)
```

Comment:
Are there reasons to use `ArrayList` instead of `List`?

Reply:
Yeah, ArrayList's constructor takes another collection as an argument and adds all of its elements into the new list. It's more efficient as by default even `addAll` still adds elements one by one, causing resizes and so on